share/python/examples/gpu_viewer.py

```diff
@@ -0,0 +1,1367 @@
+
+void main() {{
+    vec4 inColor = texture(imageTex, vert_texCoord);
+    vec4 outColor = OCIOMain(inColor);
+    frag_color = outColor;
+}}
+"""
+"""
+Fragment shader which performs a 2D texture lookup to map an image 
+texture onto UVs and processes fragments through an OCIO-provided 
+shader program segment, which itself utilizes additional texture 
+lookups, dynamic property uniforms, and various native GLSL op 
+implementations. Note that this shader's cost will increase with 
+additional LUTs in an OCIO processor, since each adds its own 
+2D or 3D texture.
+"""
+
+
+class ImagePlane(QtOpenGL.QGLWidget):
+    """
+    Qt-wrapped OpenGL window for drawing with PyOpenGL.
+    """
+
+    def __init__(self, parent=None):
+        super(ImagePlane, self).__init__(parent)
+
+        # Clicking on/tabbing to widget restores focus
+        self.setFocusPolicy(QtCore.Qt.StrongFocus)
+
+        # Set to True after initializeGL is called. Don't allow grabbing
+        # OpenGL context until that point.
+        self._gl_ready = False
+
+        # Color management
+        self._ocio_input_cs = None
+        self._ocio_display = None
+        self._ocio_view = None
+        self._ocio_exposure = 0.0
+        self._ocio_gamma = 1.0
+        self._ocio_channel_hot = [1, 1, 1, 1]
+        self._ocio_proc_cache_id = None
+        self._ocio_shader_cache_id = None
+        self._ocio_shader_desc = None
+        self._ocio_tex_start_index = 1  # Start after image_tex
+        self._ocio_tex_ids = []
+        self._ocio_uniform_ids = {}
+
+        # MVP matrix components
+        self._model_view_mat = imath.M44f()
+        self._proj_mat = imath.M44f()
+
+        # Image texture
+        self._image_tex = None
+        self._image_center = imath.V2f(0.0, 0.0)
+        self._image_pos = imath.V2f(0.0, 0.0)
+        self._image_size = imath.V2f(1.0, 1.0)
+
+        # Image plane VAO
+        self._plane_vao = None
+        self._plane_position_vbo = None
+        self._plane_tex_coord_vbo = None
+        self._plane_index_vbo = None
+
+        # GLSL shader program
+        self._vert_shader = None
+        self._frag_shader = None
+        self._shader_program = None
+
+        # Setup keyboard shortcuts
+        self._install_shortcuts()
+
+    def initializeGL(self):
+        """
+        Set up OpenGL resources and state (called once).
+        """
+        self._gl_ready = True
+
+        # Init image texture
+        self._image_tex = GL.glGenTextures(1)
+        GL.glActiveTexture(GL.GL_TEXTURE0)
+        GL.glBindTexture(GL.GL_TEXTURE_2D, self._image_tex)
+        GL.glTexImage2D(
+            GL.GL_TEXTURE_2D,
+            0,
+            GL.GL_RGBA32F,
+            self._image_size.x,
+            self._image_size.y,
+            0,
+            GL.GL_RGBA,
+            GL.GL_FLOAT,
+            ctypes.c_void_p(0)
+        )
+
+        GL.glTexParameteri(
+            GL.GL_TEXTURE_2D,
+            GL.GL_TEXTURE_WRAP_S,
+            GL.GL_CLAMP
```

Comment:
Replacing by `GL.GL_CLAMP_TO_EDGE` to fix usage under the Core Profile (`GL_CLAMP` seems to be deprecated in 3.0 and removed in 3.2).